### ISSUE TITLE
[commands]commands.Choice

### ISSUE THREAD

**Issue by sevenc-nanashi** [Rapptz/discord.py#6508]
### Summary

I want a new type for commands.

### What is the feature request for?

discord.ext.commands

### The Problem

This feature will help to make choices.
Like setting custom system message, buy/sell choice etc.

#### commands.Choice
Like typing.Union, commands.Greedy.
Can use like these:

`Choice["join", "leave"]`
`Choice["everyone", Union[discord.Member, discord.Role]]`

#### commands.errors.UnexpectedChoice
Raise when unexpected choice passed.
##### Attribute
UnexpectedChoice.expected_choices: List
`Choice["join", "leave"]` -> `["join", "leave"]`
`Choice["everyone", Union[discord.Member, discord.Role]]` -> `["everyone", Union[discord.Member, discord.Role]]`
```py
@bot.command()
async def set_sysmsg(ctx, event_name: commands.Choice["join", "leave"], *, message):
  # Set db and do something
  await ctx.send("Successfully updated system message.")
```

#### How does it works

User#0001: /set_sysmsg join Hello, {name}!
Bot#9999: Successfully updated system message.

User#0001: /set_sysmsg invalid This is invalid!
Bot#9999: commands.errors.UnexpectedChoice: Unexpected choice passed.

### The Current Solution

```py
@bot.command()
async def set_sysmsg(ctx, event_name, *, message):
  if event_name not in ("join", "leave"):
    return await ctx.send("Event name isn't valid.")
  # Set db and do something
  await ctx.send("Successfully updated system message.")
```

#### How does it works

User#0001: /set_sysmsg join Hello, {name}!
Bot#9999: Successfully updated system message.

User#0001: /set_sysmsg invalid This is invalid!
Bot#9999: Event name isn't valid.

**Comment by Gobot1234** [comment]
This should hopefully be possible in V2 with a `typing(_extensions).Literal` converter.

**Comment by sevenc-nanashi** [comment]
what is "V2"?

**Comment by mesub7** [comment]
V2 is the next major version of discord.py (after 1.7).

**Comment by Rapptz** [comment]
Fixed by #6619 and using `typing.Literal`.
